internal/runners/initialize/init.go

```diff
@@ -103,11 +106,55 @@ func prepare(params *RunParams) error {
 
 // Run kicks-off the runner.
 func (r *Initialize) Run(params *RunParams) error {
-	_, err := run(params, r.Outputer)
+	_, err := run(params, r.out, r.proj)
 	return err
 }
 
-func run(params *RunParams, out output.Outputer) (string, error) {
+func runHeadless(params *RunParams, out output.Outputer, proj *project.Project) (string, error) {
+	if params.Language != "" {
+		return "", locale.NewInputError("init_headless_lang_provided_err", "You should not provide a language, dude!")
+	}
+
+	headlessCommitID := proj.CommitID()
+
+	err := proj.Source().SetNamespace(params.Namespace.Owner, params.Namespace.Project)
+	if err != nil {
+		return "", errs.Wrap(err, "Could not set namespace in project file.")
+	}
+
+	fail := proj.Source().SetCommit(headlessCommitID, false)
+	if fail != nil {
+		return "", errs.Wrap(fail.ToError(), "Could not set commit id in project file.")
+	}
+
+	path := params.Path
+	if path == "" {
+		path, err = osutils.Getwd()
+		if err != nil {
+			return "", errs.Wrap(err, "Could not determine current working directory.")
+		}
+	}
+
+	out.Notice(locale.Tr(
+		"init_success",
+		params.Namespace.Project, params.Namespace.Owner, path))
+	return path, nil
+}
```

Comment:
I don't think this needs its own function. The logic is 90% the same, the only difference with the headless commit is that you seed a commit ID. So the logic between headless and not headless should just be the same with one single if conditional that sets the commit.

Reply:
IMO the other initialization branch is quite different, as it does not have a project file yet, so it needs to be created from scratch.  As a result, there is not need to modify the project file later (as I do in the `runHeadless` function by changing namespace and commitID here).

The only part that I would re-use is the output of the success message in the end.  But I decided that it was not worth it, to introduce a big `if headless { } else { }` block in the `run` method.  Do you want me to do that here though?